src/Framework/FieldsAPI/Concerns/HasMaxLength.php

```diff
@@ -2,39 +2,43 @@
 
 namespace Give\Framework\FieldsAPI\Concerns;
 
+use Give\Framework\Validation\Rules\Max;
+
 /**
+ * @unreleased update to new validation system
  * @since 2.14.0
- *
- * @property ValidationRules $validationRules
  */
 trait HasMaxLength
 {
-
     /**
      * Set the value’s maximum length.
      *
+     * @unreleased update to use the new validation system
      * @since 2.14.0
-     *
-     * @param int $maxLength
-     *
-     * @return $this
      */
-    public function maxLength($maxLength)
+    public function maxLength(int $maxLength): self
     {
-        $this->validationRules->rule('maxLength', $maxLength);
+        if ( $this->validationRules->hasRule('max') ) {
+            $this->validationRules->removeRuleWithId('max');
+        }
```

Comment:
Interesting 🤔

Reply:
Hmm. I guess rules with options, such as Size, Max, and Min, should probably have a way of being retrieved and modified. If someone wants to grab a field and modify a rule, it should be possible without having to replace the rule entirely (and changing the rule order). Would you agree, @jonwaldstein?

Reply:
Yeah I would agree with that.

Reply:
Resolved in 19cb33527